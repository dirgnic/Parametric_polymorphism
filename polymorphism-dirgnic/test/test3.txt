def identity[A](x : A) {
  return x;
}

def main() {
  let a = identity[Int](5);
  let b = identity[Bool](false);
  return 0;
}
