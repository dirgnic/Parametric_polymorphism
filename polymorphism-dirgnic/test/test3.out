define identity_Int(x) {
  return(x)

}

define identity_Bool(x) {
  return(x)

}

define main() {
  let a = identity_TInt(5)
  let b = identity_TBool(false)
  return(0)

}


